flink-runtime/src/main/java/org/apache/flink/runtime/operators/shipping/OutputEmitter.java

```diff
@@ -143,14 +148,25 @@ public OutputEmitter(ShipStrategyType strategy, TypeComparator<T> comparator, Pa
 	
 	// --------------------------------------------------------------------------------------------
 
+	private int[] forward() {
+		if (this.channels == null) {
```

Comment:
Yes, I did overlook that, though customPartition does likewise and the OutputEmitter will only be used for a single strategy. The strategy is set in the constructor so the channel list can be created there for the types which do not broadcast.